File: tests/pywin_client-py2many/win32com_/test/testServers.jl
module testServers
using PyCall
pythoncom = pyimport("pythoncom")
import win32com.demos.connect
using win32com.test.util: RegisterPythonServer
using win32com.servers: interp
import win32com.client.dynamic
import winerror
import win32com.test.util

abstract type AbstractInterpCase <: Abstractwin32com.test.util.TestCase end
abstract type AbstractConnectionsTestCase <: Abstractwin32com.test.util.TestCase end
function TestConnections()
    test(win32com.demos.connect)
end

mutable struct InterpCase <: AbstractInterpCase
end
function setUp(self::InterpCase)
    RegisterPythonServer(interp.__file__, "Python.Interpreter")
end

function _testInterp(self::InterpCase, interp)
    assertEqual(self, Eval(interp, "1+1"), 2)
    assertRaisesCOM_HRESULT(
        win32com.test.util,
        self,
        winerror.DISP_E_TYPEMISMATCH,
        interp.Eval,
        2,
    )
end

function testInproc(self::InterpCase)
    interp = Dispatch(
        win32com.client.dynamic,
        "Python.Interpreter",
        clsctx = pythoncom.CLSCTX_INPROC,
    )
    _testInterp(self, interp)
end

function testLocalServer(self::InterpCase)
    interp = Dispatch(
        win32com.client.dynamic,
        "Python.Interpreter",
        clsctx = pythoncom.CLSCTX_LOCAL_SERVER,
    )
    _testInterp(self, interp)
end

function testAny(self::InterpCase)
    interp = Dispatch(win32com.client.dynamic, "Python.Interpreter")
    _testInterp(self, interp)
end

mutable struct ConnectionsTestCase <: AbstractConnectionsTestCase
end
function testConnections(self::ConnectionsTestCase)
    TestConnections()
end

function main()
end

main()
end

File: tests/pywin/win32com_/test/testServers.py
import pythoncom, win32com_.client.dynamic, sys
import winerror
import win32com_.test.util
import unittest


def TestConnections():
    import win32com_.demos.connect

    win32com_.demos.connect.test()


class InterpCase(win32com_.test.util.TestCase):
    def setUp(self):
        # Ensure the correct version registered.
        from win32com_.test.util import RegisterPythonServer
        from win32com_.servers import interp

        RegisterPythonServer(interp.__file__, "Python.Interpreter")

    def _testInterp(self, interp):
        self.assertEqual(interp.Eval("1+1"), 2)
        win32com_.test.util.assertRaisesCOM_HRESULT(
            self, winerror.DISP_E_TYPEMISMATCH, interp.Eval, 2
        )

    def testInproc(self):
        interp = win32com_.client.dynamic.Dispatch(
            "Python.Interpreter", clsctx=pythoncom.CLSCTX_INPROC
        )
        self._testInterp(interp)

    def testLocalServer(self):
        interp = win32com_.client.dynamic.Dispatch(
            "Python.Interpreter", clsctx=pythoncom.CLSCTX_LOCAL_SERVER
        )
        self._testInterp(interp)

    def testAny(self):
        interp = win32com_.client.dynamic.Dispatch("Python.Interpreter")
        self._testInterp(interp)


class ConnectionsTestCase(win32com_.test.util.TestCase):
    def testConnections(self):
        TestConnections()


if __name__ == "__main__":
    unittest.main("testServers")
